components/Templates/includes/auto-template/Pods_Templates_Auto_Template_Front_End.php

```diff
@@ -231,95 +267,202 @@ public function auto_pods() {
+					if (
+						$current_filter === $possible_pod_data['single_filter'] ||
+						$current_filter === $possible_pod_data['archive_filter']
+					) {
+						$pod_id   = (int) $obj;
+						$pod_name = $possible_pod;
+						$pod_type = $possible_pod_data['type'];
+					}
+				}
+			} else {
+				$obj = null;
+			}
+		}
+
+		if ( ! $pod_id ) {
+			// Build Pods object for current item.
+			$pod_id = get_queried_object_id();
+			if ( is_singular() || in_the_loop() ) {
+				$pod_type = 'post';
+			} elseif ( is_tax() ) {
+				// Outside the loop in a taxonomy, we want the term.
+				$obj      = get_queried_object();
+				$pod_type = 'taxonomy';
+				$pod_name = $obj->taxonomy;
+			} else {
+				// Backwards compatibility.
+				global $post;
+				if ( $post ) {
+					$pod_id   = $post->ID;
+					$pod_type = 'post';
+				}
+			}
+		}
+		// @todo Users?
+		// @todo Media?
+
+		// Check if $current_post_type is the key of the array of possible pods.
+		if ( $pod_type && isset( $possible_pods[ $pod_name ] ) ) {
+			// Get array for the current post type.
+			$auto_pod = $possible_pods[ $pod_name ];
 
-			$filter = $this->get_pod_filter( $current_post_type, $possible_pods );
+			$filter = $this->get_pod_filter( $pod_name, $possible_pods );
 
 			if ( current_filter() !== $filter ) {
 				return $content;
 			}
 
-			if ( ! in_the_loop() && ! pods_v( 'run_outside_loop', $this_pod, false ) ) {
-				// If outside of the loop, exit quickly
+			if ( 'post' === $pod_type && ! in_the_loop() && ! pods_v( 'run_outside_loop', $auto_pod, false ) ) {
+				// If outside of the loop, exit quickly.
 				return $content;
 			}
 
-			// build Pods object for current item
-			global $post;
-			$pods = pods( $current_post_type, $post->ID );
-
-			if ( $this_pod['single'] && is_singular( $current_post_type ) ) {
-				// load the template
-				$content = $this->load_template( $this_pod['single'], $content, $pods, $this_pod['single_append'] );
-			} //end if
-			// check if we are on an archive of the post type
-			elseif ( $this_pod['archive'] && is_post_type_archive( $current_post_type ) ) {
-				// load the template
-				$content = $this->load_template( $this_pod['archive'], $content, $pods, $this_pod['archive_append'] );
-
-			} elseif ( is_home() && $this_pod['archive'] && $current_post_type === 'post' ) {
-				// if pfat_archive was set and we're in the blog index, try to append template
-				// append the template
-				$content = $this->load_template( $this_pod['archive'], $content, $pods, $this_pod['archive_append'] );
-
-			} elseif ( is_tax( $current_post_type ) ) {
-				// if is taxonomy archive of the selected taxonomy
-				// if pfat_single was set try to use that template
-				if ( $this_pod['archive'] ) {
-					// append the template
-					$content = $this->load_template( $this_pod['archive'], $content, $pods, $this_pod['archive_append'] );
-				}
+			$pod_name_and_id = array( $pod_name, $pod_id );
```

Comment:
:no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: d20c543a192fca8b04502f7b26338ca9</sub>